### ISSUE TITLE
[Enhancement] Named snapshots

### ISSUE THREAD

**Issue by Vinschni** [gristlabs/grist-core#486]
When working in a team it would be helpful to:

1. Intentionally create a snapshot in document history
2. Name that snapshot to ease identification

This would require an  `Create Snapshot` button and a "Rename Snapshot" Entry in the context menu of the snapshots.
![Unbenannt](https://user-images.githubusercontent.com/340731/230898627-9ea1a842-2755-4245-b5cd-5b5480b0b680.png)

**Comment by paulfitz** [comment]
Hi @Vinschni, it is a good idea, and Grist does internally have the ability to put labels on snapshots. All that is really missing is a UI for it.

**Comment by softwareguru90** [comment]
Hello @paulfitz I have a requirement for this feature, too. I am trying to develop renamed snapshot myself. I got some point but I am trying to understand that is there a way how to update this [metadata label](https://github.com/gristlabs/grist-core/blob/main/app/common/DocSnapshot.ts#L18) on the model.

**Comment by paulfitz** [comment]
Hi @softwareguru90, you can set it while creating a snapshot, see: https://github.com/gristlabs/grist-core/blob/3e0045db4580f1bce49dec59afb8496024358b58/app/server/lib/ExternalStorage.ts#L28-L29. With existing code, that save happens in `HostedStorageManager.ts`, in the `_pushToS3` method https://github.com/gristlabs/grist-core/blob/3e0045db4580f1bce49dec59afb8496024358b58/app/server/lib/HostedStorageManager.ts#L750. But that doesn't really help you. I can see how setting it after the snapshot already exists would be preferable. One way to go about it would be to add a method to `ExternalStorage` (and its implementations), and to `DocSnapshotInventory`, then an API endpoint to actually call them.

**Comment by softwareguru90** [comment]
Hi @paulfitz. Thanks four your response.Snapshot section is quite complex. I am having trouble understanding exactly where to update the label field. Could you briefly explain the steps I need to follow?

**Comment by EduBic** [comment]
After 3 years, is this feature available? At least on the REST API? Not necessarily in the UI.

**Comment by paulfitz** [comment]
> After 3 years, is this feature available? At least on the REST API? Not necessarily in the UI.

No, not done! There should be enough info in this thread for someone motivated to implement though.
